etna/analysis/outliers/density_outliers.py

```diff
@@ -128,8 +128,8 @@ def get_anomalies_density(
     for seg in segments:
         # TODO: dropna() now is responsible for removing nan-s at the end of the sequence and in the middle of it
         #   May be error or warning should be raised in this case
-        segment_df = ts[:, seg, :][seg].dropna().reset_index()
-        series = segment_df[in_column].values
+        segment_df = ts[:, seg, :][seg].reset_index()
```

Comment:
This was a very strange place for an error. In the old version it removes rows even if NaNs in non-relevant features. It became an issue when I rewrote `OutliersTransform.fit`.

Reply:
I'm also confused that we handle this differently in all our anomaly detection methods.

Reply:
In case of density outliers raising error for NaNs seems like valid decision.
But maybe we shold do this updates in scope of all methods.

Reply:
Let's not fix this in this particular task.